terra_notebook_utils/drs.py

```diff
@@ -121,6 +125,25 @@ def copy_to_local(drs_url: str,
         logger.info(f"Downloading {drs_url} to {filepath}")
         blob.download_to_file(fh)
 
+def head(drs_url: str,
+         n: int = 1,
+         workspace_name: Optional[str]=WORKSPACE_NAME,
+         google_billing_project: Optional[str]=WORKSPACE_GOOGLE_PROJECT):
+    """
+    Check access to a DRS object by attempting to access its first byte of data.
+    """
+    assert drs_url.startswith("drs://")
+    enable_requester_pays(workspace_name, google_billing_project)
+    client, info = resolve_drs_for_gs_storage(drs_url)
+    blob = client.bucket(info.bucket_name, user_project=google_billing_project).blob(info.key)
+    try:
+        # don't expand compressed files, to save time
+        return blob.download_as_bytes(start=0, end=n, raw_download=True)
+    except Exception:
```

Comment:
This clause should catch exceptions related to an inaccessible object. Other exceptions should be raised normally.

Please consider:
```
except (google.cloud.exceptions.NotFound, google.cloud.exceptions.Forbidden):
```